pkg/agent/plugin/workloadattestor/docker/docker.go

```diff
@@ -64,11 +63,8 @@ type dockerPluginConfig struct {
 	DockerSocketPath string `hcl:"docker_socket_path"`
 	// DockerVersion is the API version of the docker daemon (default: "1.40").
 	DockerVersion string `hcl:"docker_version"`
-	// CgroupPrefix is the cgroup prefix to look for in the cgroup entries (default: "/docker").
-	CgroupPrefix string `hcl:"cgroup_prefix"`
-	// CgroupContainerIndex is the index within the cgroup path where the container ID should be found (default: 1).
-	// This is a *int to allow differentiation between the default int value (0) and the absence of the field.
-	CgroupContainerIndex *int `hcl:"cgroup_container_index"`
+	// ContainerIDCGroupMatchers
+	ContainerIDCGroupMatchers []string `hcl:"container_id_cgroup_matchers"`
```

Comment:
Did the prefix stuff make it into a release? If so, do we still want to provide a measure of back-compat even though it was undocumented.... @evan2645, thoughts?

Reply:
It's relatively old, I'm sure it made it to a release. I'm happy to hack something together as needed, given its undocumented I had my fingers crossed I could avoid that :)

Reply:
Evan and I had a chat offline. I think we'd both be more comfortable if we preserved backwards compatibility, including the prefix stuff.

Reply:
I'll take a pass at that. I think the best I can do is support the previous configurables but assume the prefix is at least full path segments. For example

```
prefix: /system.slice
index: 3
```

would be
```
/system.slice/*/*/<id>
```

```
prefix: /system.slice/container-
index: 3
```

would be
```
/system.slice/container-/*/<id>
```

i.e. if you were previously configuring a prefix that matched a partial segment, that won't be supported.

Reply:
rather than converge the old configurables onto the new library I retained the old code as well. This ensures they will continue to work as they did before but its not my favourite.

Ideally, we delete those configurables and code for 0.9? Does it have to wait until 0.10/1.0?

Reply:
We'll need to leave it deprecated for at one release. Since we haven't cut 0.9 yet, we can probably remove it from master as soon as 0.9 is released.

Reply:
yup, make sense.